### PR TITLE
Display trustline authorization state in the balances list 

### PR DESCRIPTION
**Pull request by r8d8**
Related to #159

### CODE REVIEW

File: services/horizon/internal/resource/account.go [hunk 1]
```diff
@@ -89,3 +100,20 @@ func (this *Account) MustGetData(key string) []byte {
 func (this *Account) GetData(key string) ([]byte, error) {
 	return base64.StdEncoding.DecodeString(this.Data[key])
 }
+
+/// splitTrustlines splits trustlines into authorized/unauthorized arrays
```
Comment (dulanov) [inline services/horizon/internal/resource/account.go]: Wrong comment here - `///`.

File: services/horizon/internal/resource/account.go [hunk 2]
```diff
@@ -89,3 +100,20 @@ func (this *Account) MustGetData(key string) []byte {
...
+func splitTrustlines(ct []core.Trustline) (authorized, unauthorized []core.Trustline) {
+	authorized = make([]core.Trustline, 0)
+	unauthorized = make([]core.Trustline, 0)
+
+	for _, tl := range ct {
+		switch tl.Flags {
+		case 1:
```
Comment (bartekn) [inline services/horizon/internal/resource/account.go]: New flag can be added in a future. You should use `xdr.TrustLineFlagsAuthorizedFlag` as a mask.

File: services/horizon/internal/resource/account.go [hunk 3]
```diff
@@ -89,3 +100,20 @@ func (this *Account) MustGetData(key string) []byte {
...
+			authorized = append(authorized, tl)
+		case 0:
+			unauthorized = append(unauthorized, tl)
+		}
```
Comment (bartekn) [inline services/horizon/internal/resource/account.go]: New flag can be added in a future. You should use `xdr.TrustLineFlagsAuthorizedFlag` as a mask.

File: services/horizon/internal/resource/account.go [hunk 4]
```diff
@@ -30,15 +30,26 @@ func (this *Account) Populate(
...
+	this.Balances = make([]Balance, len(authTl)+1)
+	for i, tl := range authTl {
 		err = this.Balances[i].Populate(ctx, tl)
 		if err != nil {
 			return
 		}
 	}
 
+	// populate unauthorized balances
+	this.UnauthorizedBalances = make([]Balance, len(unauthTl)+1)
```
Comment (bartekn) [inline services/horizon/internal/resource/account.go]: We don't need `+1` here. It's only needed for XLM balance and it's not a trust line.

File: services/horizon/internal/resource/account.go (hunk 3 above)
Comment (r8d8) [reply services/horizon/internal/resource/account.go]: Ok, thanks @bartekn

File: services/horizon/internal/resource/account.go [hunk 5]
```diff
@@ -89,3 +101,19 @@ func (this *Account) MustGetData(key string) []byte {
...
 	return base64.StdEncoding.DecodeString(this.Data[key])
 }
+
+// splitTrustlines splits trustlines into authorized/unauthorized arrays
+func splitTrustlines(ct []core.Trustline) (authorized, unauthorized []core.Trustline) {
+	authorized = make([]core.Trustline, 0)
+	unauthorized = make([]core.Trustline, 0)
+
+	for _, tl := range ct {
+		if  tl.Flags == int32(xdr.TrustLineFlagsAuthorizedFlag) {
```
Comment (bartekn) [inline services/horizon/internal/resource/account.go]: This needs to be changed to something like:
```go
if tl.Flags & int32(xdr.TrustLineFlagsAuthorizedFlag) != 0 {
```

**Comment by r8d8** [comment]
@bartekn, I added test for trustllline splitting.
Currenlty can't see any tests for `balance`, and maybe it will require some refactoring to provide them.

File: services/horizon/internal/resource/account.go [hunk 6]
```diff
@@ -89,3 +101,19 @@ func (this *Account) MustGetData(key string) []byte {
...
 	return base64.StdEncoding.DecodeString(this.Data[key])
 }
+
+// splitTrustlines splits trustlines into authorized/unauthorized arrays
+func splitTrustlines(ct []core.Trustline) (authorized, unauthorized []core.Trustline) {
+	authorized = make([]core.Trustline, 0)
+	unauthorized = make([]core.Trustline, 0)
+
+	for _, tl := range ct {
+		if tl.Flags & int32(xdr.TrustLineFlagsAuthorizedFlag) != 0 {
```
Comment (nullstyle) [inline services/horizon/internal/resource/account.go]: Please extract to a method on `core.Trustline` named `IsAuthorized`

**Comment by tomerweller** [comment]
What's the status of this?

**Comment by howardtw** [comment]
Closing as this is being replaced by https://github.com/stellar/go/pull/1011

**Pull request closed without merging**